### ISSUE TITLE
Recurring Job triggers on Update

### ISSUE THREAD

**Issue by twojnarowski** [HangfireIO/Hangfire#2381]
The situation sounds similar to: https://github.com/HangfireIO/Hangfire/issues/728 and https://github.com/HangfireIO/Hangfire/issues/1797

I am using Hangfire 1.8.10 currently and I had a bit similar situation. However I haven't changed cron, I changed the job itself. This is what happened:
- I had a recurring job that used a method.
- I have refactored that method, as well as the code that creates this recurring job via RecurringJobManager
- I didn't think about the fact that I should somehow tell Hangfire that it should run a different method now
- I pushed the changes to a staging environment
- When the time came, hangfire couldn't run the job and produced a failed state "Can not find the target method" (that is of course correct and expected)
- I added code so that when the application starts, it uses the `recurringJobManager.AddOrUpdate` method to update each job in case of code change
- I updated the application on the server, and when it started, one job (which worked after 1st update) was updated and did not run. The second job (the one that failed) was triggered and executed right after the AddOrUpdate. (the cron wasn't changed as in the mentioned older issues).
- After that, I prepared another application update (nothing related to Hangfire) and after updating the application, which executed the AddOrUpdate for each job that prevents "Can not find the target method" after a code change, the outcome is that the job did not trigger. 

I have no idea right now how to verify whether the job running was caused because the job failed and didn't run, so the last execution time was over 24 hours ago, so it did trigger to run that failed execution "ASAP" (and that is a feature / expected behavior) or was it triggered as a bug, just because the insides of the recurring job changed...

**Comment by odinserj** [comment]
Thank you for the detailed description. I have just completed refactoring for recurring jobs to make them look clean, simple and avoid side effects. As a bonus result, the described problem is gone 😊

**Comment by twojnarowski** [comment]
I just noticed something as a follow-up to this topic, but I'm not sure if that's an issue or a feature.
I am now on 1.8.10 of Hangfire. When I have posted an update on the server, refactoring some jobs, then my code that runs for each recurring job AddOrUpdates at the start of the app, everything works, but in the Jobs listing, these jobs that have been modified, don't show the name of the function, but "Can not find the target method" instead. 
So When I go into details of a Job, under State I see "Succeeded" but on top of the page I get a yellow warning saying "Can not find the target method." 
So just the displaying of the name of the functions that were run when the function was in a different file doesn't work.

I will update to 1.8.12 soon and see if the history of jobs still gets corrupted after a job is changed.
